extensions/EntityExtension/src/main/kotlin/com/typewritermc/entity/entries/activity/AmbientSoundActivity.kt

```diff
@@ -64,7 +65,7 @@ class AmbientSoundActivity(
             nextPlay = System.currentTimeMillis() + delay.random().toMillis()
             val sound = sounds.weightedRandom().sound
             context.viewers.forEach { viewer ->
-                sound.play(viewer)
+                sound.play(viewer, viewer.interactionContext ?: return@forEach)
```

Comment:
The return here is not okay. As the interaction context can definitely be null it will break and not play a sound if the user is not in an interact. You should refactor to allow for a null value of the context. It will accept a null context